### ISSUE TITLE
AWS VM maintenance tasks

### ISSUE THREAD

**Issue by GINAGC** [k8-proxy/GW-Releases#72]
Placeholder to track the work done on AWS for VM maintenance.

**Comment by faisaladnanpeltops** [comment]
Update:
- Access to AWS Config and Lambda is granted
- Testing boto3 
- Testing elasticsearch

**Comment by faisaladnanpeltops** [comment]
Update:

- [x] Access to AWS Config and Lambda is granted
- [x] Testing boto3
- [x] Testing elasticsearch

**Comment by faisaladnanpeltops** [comment]
Next step:
[] Create lambda function that sample AWS Config every minutes
[] Create dashboard that shows running and stopped instances

**Comment by faisaladnanpeltops** [comment]
Update

Next
- [x] Create lambda function that sample AWS Config every minutes
- [x] Create dashboard that shows running and stopped instances

Create dashboard that show this information:
3.- Automate daily report with the following information in dashboards (to be sent daily at the end of the day to slack):
3.1.- Running VMs by tag
3.2.- New VMs of the day
3.3.- Deleted VMs on the day
3.4.- Total of running and shutdown VMs (running status)
3.5.- Ports opened by VM
3.6.- IP address by VM

**Comment by aduvukace** [comment]
@GINAGC this task is complete

**Comment by faisaladnanpeltops** [comment]
@aduvukace , @GINAGC ,
the first dashboard draft is ready, and the additional dashboards below is still ongoing:
Should I continue with dashboard below?

3.- Automate daily report with the following information in dashboards (to be sent daily at the end of the day to slack):
3.1.- Running VMs by tag
3.2.- New VMs of the day
3.3.- Deleted VMs on the day
3.4.- Total of running and shutdown VMs (running status)
3.5.- Ports opened by VM
3.6.- IP address by VM

**Comment by GINAGC** [comment]
@faisaladnanpeltops yes, but first send the dashboard link and access to the channel. This is P2, so after any P1 task.

**Comment by faisaladnanpeltops** [comment]
@GINAGC , yes, sure. I'm pausing the work on this issue.

**Comment by gurunathg** [comment]
Updates on maintenance tasks:

1. Config rule to monitor mandatory tags have been deployed to all regions
2. List of non-compliant instances has been shared with everyone and sent a notification. These will be stopped EOD. Starting from tomorrow, all non-compliant instances will be stopped automatically.
3. All instances marked as "Delete:Yes" has been terminated programmatically. To deploy this to lambda, I will request for role to terminate instances. Starting from tomorrow, all instances marked as "Delete:Yes" will be terminated at starting of each day

@GINAGC Can you please let me know if there is any other thing from my side on this task or this can be closed

**Comment by GINAGC** [comment]
@gurunathg we have a list of metrics that the team wanted to display. So this will be more of an ongoing task, not a high priority at this moment.
